File: AndroidFeatures/app/src/main/java/com/clear/androidfeatures/jobscheduler/activities/JobIntentServiceActivity.java
package com.clear.androidfeatures.jobscheduler.activities;

import android.app.Activity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.clear.androidfeatures.R;
import com.clear.androidfeatures.jobscheduler.services.TestJobIntentService;

/**
 * Created by josegsp on 8/14/17.
 */

public class JobIntentServiceActivity extends Activity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_jobintentservice);
    }

    @Override
    public void onClick(View view) {
        switch (view.getId()) {
            case R.id.enqueue_jobintentservice: {
                TestJobIntentService.enqueue(getBaseContext());
                break;
            }

            case R.id.enqueue_jobintentservice_delayed: {
                TestJobIntentService.enqueueDelayed(getBaseContext());
                break;
            }
        }
    }
}

File: AndroidFeatures/app/src/main/java/com/clear/androidfeatures/jobintentservice/JobIntentServiceActivity.kt
package com.clear.androidfeatures.jobintentservice

import android.os.Bundle
import android.support.annotation.MainThread
import android.support.v7.app.AppCompatActivity
import com.clear.androidfeatures.R

@MainThread
class JobIntentServiceActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_job_intent_service)
    }
}